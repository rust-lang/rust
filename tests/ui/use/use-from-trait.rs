use Trait::foo; //~ ERROR `foo` is not directly importable
use Trait::Assoc; //~ ERROR `Assoc` is not directly importable
use Trait::C; //~ ERROR `C` is not directly importable

use Foo::new; //~ ERROR unresolved import `Foo` [E0432]

use Foo::C2; //~ ERROR unresolved import `Foo` [E0432]

pub trait Trait {
    fn foo();
    type Assoc;
    const C: u32;
}

struct Foo;

impl Foo {
    fn new() {}
    const C2: u32 = 0;
}

fn main() {}
