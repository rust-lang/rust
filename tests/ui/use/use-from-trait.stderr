error[E0253]: `foo` is not directly importable
  --> $DIR/use-from-trait.rs:1:5
   |
LL | use Trait::foo;
   |     ^^^^^^^^^^ cannot be imported directly

error[E0253]: `Assoc` is not directly importable
  --> $DIR/use-from-trait.rs:2:5
   |
LL | use Trait::Assoc;
   |     ^^^^^^^^^^^^ cannot be imported directly

error[E0253]: `C` is not directly importable
  --> $DIR/use-from-trait.rs:3:5
   |
LL | use Trait::C;
   |     ^^^^^^^^ cannot be imported directly

error[E0432]: unresolved import `Foo`
  --> $DIR/use-from-trait.rs:5:5
   |
LL | use Foo::new;
   |     ^^^ `Foo` is a struct, not a module

error[E0432]: unresolved import `Foo`
  --> $DIR/use-from-trait.rs:7:5
   |
LL | use Foo::C2;
   |     ^^^ `Foo` is a struct, not a module

error: aborting due to 5 previous errors

Some errors have detailed explanations: E0253, E0432.
For more information about an error, try `rustc --explain E0253`.
